File: python_files/raylib/snake/food.py
import pyray as p


class Food:
    def __init__(self, grid, snake_list) -> None:
        self.grid = grid
        self.snake_list = snake_list
        self.x, self.y = self.gen_random_food()
        self.width = self.grid.block_size
        self.height = self.grid.block_size
        self.color = p.RED
        
    def gen_random_food(self):
        while True:
            x = p.get_random_value(0, self.grid.cols -1) * self.grid.block_size + self.grid.margin
            y = p.get_random_value(0, self.grid.rows -1) * self.grid.block_size + self.grid.margin
            
            if (x, y) not in self.snake_list:
                return (x, y)
            
    def draw(self):
        p.draw_rectangle_rec((self.x, self.y, self.width, self.height), self.color)

File: c_files/raylib/snake/food.c
#include "food.h"
#include "constants.h"
#include "raylib.h"
#include "snake.h"

void initFood(Food *food, Snake *snake) {
  food->color = RED;
  genRandomFood(food, snake);
}

void genRandomFood(Food *food, Snake *snake) {
  while (1) {
    int x = GetRandomValue(0, ROWS - 1) * BLOCK_SIZE + MARGIN;
    int y = GetRandomValue(0, COLS - 1) * BLOCK_SIZE + MARGIN;

    for (int i = 0; i < snake->size; i++) {
      if (x != snake->list[i].x && y != snake->list[i].y) {
        food->x = x;
        food->y = y;
        return;
      }
    }
  }
}

void drawFood(Food *food) {
  DrawRectangleRec((Rectangle){food->x, food->y, BLOCK_SIZE, BLOCK_SIZE},
                   food->color);
}
